core/src/sample_performance_service.rs

```diff
@@ -0,0 +1,102 @@
+use solana_ledger::blockstore::Blockstore;
+use solana_ledger::blockstore_meta::PerfSample;
+use solana_runtime::bank_forks::BankForks;
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, RwLock,
+    },
+    thread::{self, sleep, Builder, JoinHandle},
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+const SAMPLE_INTERVAL: u64 = 60;
+const SLEEP_INTERVAL: u64 = 500;
+
+pub struct SamplePerformanceSnapshot {
+    pub num_transactions: u64,
+    pub num_slots: u64,
+}
+
+pub struct SamplePerformanceService {
+    thread_hdl: JoinHandle<()>,
+}
+
+impl SamplePerformanceService {
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new(
+        bank_forks: &Arc<RwLock<BankForks>>,
+        blockstore: &Arc<Blockstore>,
+        exit: &Arc<AtomicBool>,
+    ) -> Self {
+        let exit = exit.clone();
+        let blockstore = blockstore.clone();
+        let bank_forks = bank_forks.clone();
+
+        info!("Starting SamplePerformance service");
+        let thread_hdl = Builder::new()
+            .name("sample-performance".to_string())
+            .spawn(move || {
+                Self::run(bank_forks, &blockstore, exit);
+            })
+            .unwrap();
+
+        Self { thread_hdl }
+    }
+
+    pub fn run(
+        bank_forks: Arc<RwLock<BankForks>>,
+        blockstore: &Arc<Blockstore>,
+        exit: Arc<AtomicBool>,
+    ) {
+        let mut sample_snapshot: Option<SamplePerformanceSnapshot> = None;
+
+        loop {
+            if exit.load(Ordering::Relaxed) {
+                break;
+            }
+
+            let start = SystemTime::now();
```

Comment:
Why did you decide to use SystemTime? Seems a bit more complicated than just using Instant. Is it more precise in some way?

Reply:
I would imagine something more like this:
```
let mut now = Instant::now();
loop {
  if exit.load(Ordering::Relaxed) {
    break;
  }
  let elapsed = now.elapsed();
  if elapsed.as_secs() >= 60 {
    now = Instant::now();
    ...
    // pass sample_period_secs: elapsed
  }
  sleep(Duration::from_millis(SLEEP_INTERVAL));
}
```

Reply:
I'm not sure if this is necessary, but I wanted to align it to the system clock, so samples were taken exactly on the minute. Thoughts?